### PR TITLE
LHE interface for Herwig++

### PR DESCRIPTION
**Pull request by konush**

### CODE REVIEW

**Comment by ktf** [comment]
please test

**Comment by bendavid** [comment]
Still not retested...

**Comment by nclopezo** [comment]
please test

**Comment by bendavid** [comment]
Looking at the code I have the impression that all that is needed is to comment out the two LogError statements where the exceptions are caught.  Then the event will simply be skipped.  If this happens at the 1 in 10,000 level it should not cause any problems.

File: GeneratorInterface/ThePEGInterface/plugins/ThePEGHadronizer.cc [hunk 1]
```diff
@@ -105,6 +131,46 @@ bool ThePEGHadronizer::generatePartonsAndHadronize()
...
+
+	flushRandomNumberGenerator();
+	
+	//need to copy lhe event here unfortunately because of interface mismatch
+	proxy_->loadEvent(boost::shared_ptr<lhef::LHEEvent>(new lhef::LHEEvent(*lheEvent())));
+
+	try {
+		thepegEvent = eg_->shoot();
+	} catch (std::exception& exc) {
+		edm::LogError("Generator|ThePEGHadronizer") << "EGPtr::shoot() thrown an exception, event skipped: " << exc.what();
```
Comment (bendavid) [inline GeneratorInterface/ThePEGInterface/plugins/ThePEGHadronizer.cc]: Would simply comment out this line

File: GeneratorInterface/ThePEGInterface/plugins/ThePEGHadronizer.cc [hunk 2]
```diff
@@ -105,6 +131,46 @@ bool ThePEGHadronizer::generatePartonsAndHadronize()
...
+		return false;
+	} catch (...) {
+		edm::LogError("Generator|ThePEGHadronizer") << "EGPtr::shoot() thrown an unknown exception, event skipped";
```
Comment (bendavid) [inline GeneratorInterface/ThePEGInterface/plugins/ThePEGHadronizer.cc]: and this one

**Comment by bendavid** [comment]
-1
This can be closed in favour of https://github.com/cms-sw/cmssw/pull/7143

**Pull request closed without merging**